core/raw/metadata.cpp

```diff
@@ -17,121 +17,143 @@
+    {"T", "T", averageMode::AVGE},              {"teload", "teload", averageMode::AVGE},
+    {"tepos", "tepos", averageMode::AVGE},      {"teext", "teext", averageMode::AVGE},
+    {"xe", "xe", averageMode::AVGE},            {"ye", "ye", averageMode::AVGE},
+    {"ze", "ze", averageMode::AVGE},            {"mon", "mon", averageMode::LAST},
+    {"delta_mon", "Δmon", averageMode::SUM},    {"t", "t", averageMode::LAST},
+    {"delta_t", "Δt", averageMode::SUM},        {"date", "date", averageMode::FIRST},
+    {"comment", "comment", averageMode::FIRST},
+};
 
-int Metadata::noNumAttr = 0;
-std::vector<MetaDefinition> Metadata::metaKeys_ = Metadata::metaKeys();
+} // namespace
 
-MetaDefinition::MetaDefinition(const QString& niceName, averageMode avgmode, valueType valtype)
-    : name{niceName}
+MetaDefinition::MetaDefinition(const QString& name, const QString& niceName, averageMode avgmode)
+    : niceName{niceName}
+    , asciiName{name}
     , mode{avgmode}
-    , type{valtype}
 {
-    if (valtype == valueType::STRING)
-        Metadata::noNumAttr++;
+    if (mode == averageMode::FIRST)
+        noNumAttr++;
 }
 
 Metadata::Metadata()
     : Mapped{}
 {}
 
-int Metadata::numAttributes(bool onlyNum)
+QString Metadata::attributeStrValue(int i) const
 {
-    return onlyNum ? metaKeys_.size()-noNumAttr : metaKeys_.size();
+    QVariant v = attributeValue(i);
+    if (v.canConvert<deg>())
+        return QString::number(double(v.value<deg>()));
+    else
+        return v.toString();
 }
 
-const QString& Metadata::attributeTag(int i, bool nice)
+QVariant Metadata::attributeValue(int i) const
 {
-    return attributeTags(nice).at(i);
+    if (i >= metaDefs_.size())
+        qFatal("impossible case");
+    else
+        return at(metaDefs_.at(i).asciiName);
 }
 
-const QStringList& Metadata::attributeTags(bool nice)
+std::vector<QVariant> Metadata::attributeValues() const
 {
-    return nice ? niceTags : asciiTags;
+    std::vector<QVariant> attrs;
+    for (int i=0; i<metaDefs_.size(); ++i)
+        attrs.push_back(attributeValue(i));
+    return attrs;
 }
 
-QString Metadata::attributeStrValue(int i) const
+namespace meta {
+
+QStringList asciiTags;
+QStringList niceTags;
+
+int size()
 {
-    return attributeValue(i).toString();
+    return attributeNaNs().size();
 }
 
-QVariant Metadata::attributeValue(int i) const
+int numAttributes(bool onlyNum)
 {
-    if (i >= metaKeys_.size())
-        qFatal("impossible case");
-    else {
-        MetaDefinition metaKey = metaKeys_.at(i);
-        if (metaKey.type == valueType::DEG)
-            return QVariant(at<deg>(metaKey.name));
-        else if (metaKey.type == valueType::STRING)
-            return at<QString>(metaKey.name);
-        else
-            return at<double>(metaKey.name);
-    }
+    return onlyNum ? metaDefs_.size()-noNumAttr : metaDefs_.size();
 }
 
-std::vector<QVariant> Metadata::attributeValues() const
+const QString& attributeTag(int i, bool nice)
```

Comment:
bitte auflösen in zwei Funktionen, asciiTag(i) und niceTag(i).

und vielleicht nicht diese Funktionen mittels ...Tags() implementieren, sondern gerade anders herum